sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -85,6 +79,60 @@ def _find_binning_thresholds(data, max_bins, subsample, random_state):
     return binning_thresholds
 
 
+def _find_categories(data, max_bins, is_categorical):
+    """Extract feature-wise categories from categorical data
+
+    Missing values and negative values are ignored. They will be considered
+    missing when ``_encode_categories`` is called.
+
+    Parameters
+    ----------
+    data : ndarray of shape (n_samples, n_features)
+        The data to bin.
+    max_bins: int
+        The maximum number of bins to use for non-missing values. If for a
+        given feature the number of unique values is less than ``max_bins``,
+        then those unique values will be used to compute the bin thresholds,
+        instead of the quantiles
+    is_categorical : ndarray of bool of shape (n_features,) or None
+        Indicates categorical features.
+
+    Return
+    ------
+    bin_categories_ : dict of int to arrays
+        For each categorical feature, this gives a maps categorical indicies
+        to the categories corresponding to each bin.
```

Comment:
Is this more correct? I don't understand what a "categorical index" is.

There are other places where this docstring is copy/paste.


```suggestion
    bin_categories_ : dict of int to arrays
        For each categorical feature, this gives a map from bin index to categorical value.
        The size of each array is max_bins or less depending on the number of categories.
```